### ISSUE TITLE
Installing docker within a docker container

### ISSUE THREAD

**Issue by ColinHebert** [garethr/garethr-docker#399]
Currently this puppet module for docker can't be used within a docker container (to install docker-engine).

First, it relies on `apt-transport-https` which may not be installed, and installing it would require running `apt-get update` beforehand which will very likely end up in a loop (as we expect to run apt-get update only after setting up our HTTPs repositories).
The workaround here is to either install `apt-transport-https` earlier (I went with this solution), or running an `apt-get update` before the package gets installed (but not using the one from the apt module).

The second issue is that the module makes the assumption that the goal is to set up the docker service instead of simply setting up the docker client.

In https://github.com/garethr/garethr-docker/blob/2d050ec40a790d5a360ffb04d9f435bb00858583/manifests/params.pp#L59 in particular, the `params.pp` calls systemd_reload which fails within a docker container. This step isn't necessary when attempting to install the docker tools rather than trying to run the daemon right away.

It'd be nice to have access to a "no-daemon" option or something along those lines to set up docker without doing anything with the daemon (ie, if you want to install docker to manage a remote docker swarm, or something like this).

**Comment by garethr** [comment]
Thanks for the issues. #369 is tracking some of the issues with apt-transport-https. I'll close this in favour of tracking everything there.

Master should now have a number of options including the ability to set `manage_service` to `false` which should do what you think (I think).
